src/include/duckdb/optimizer/join_order/estimated_properties.hpp

```diff
@@ -45,4 +48,10 @@ double EstimatedProperties::GetCardinality() const;
 template <>
 idx_t EstimatedProperties::GetCardinality() const;
 
+template <>
+double EstimatedProperties::GetCost() const;
+
+template <>
+idx_t EstimatedProperties::GetCost() const;
```

Comment:
Not related to this PR - but I think we should remove the `double` everywhere and only use `idx_t` for cardinalities/cost.

Reply:
If I remember correctly, rounding `double`s to `idx_t` resulted in worse plans